File: backend/src/main/java/ropold/backend/model/UserModel.java
package ropold.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "users")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserModel {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(columnDefinition = "UUID")
    private UUID id;

    @Column(name = "microsoft_id", unique = true, nullable = false)
    private String microsoftId;

    @Column(name = "username", nullable = false)
    private String username;

    @Column(name = "email", unique = true, nullable = false)
    private String email;

    @Column(name = "role", nullable = false, length = 50)
    private String role;

    @Column(name = "preferred_language", length = 2)
    private String preferredLanguage = "de";

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "last_login_at")
    private LocalDateTime lastLoginAt;

    @Column(name = "avatar_url", length = 500)
    private String avatarUrl;

    @PrePersist
    protected void onCreate() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
    }


}

File: frontend/src/components/model/UserModel.ts

export type UserModel = {
    id: string;
    microsoftId: string;
    username: string;
    email: string;
    role: string;
    preferredLanguage: string;
    createdAt: string;
    lastLoginAt?: string;
    avatarUrl?: string;
}

export const DefaultUser: UserModel = {
    id: "0",
    microsoftId: "0",
    username: "Loading...",
    email: "Loading...",
    role: "USER",
    preferredLanguage: "de",
    avatarUrl: undefined,
    lastLoginAt: undefined,
    createdAt: "Loading..."
};